### PR TITLE
[rz-diff] Optimize diffing functions.

### LINKED ISSUE #1044: Endless 'Exceeded analysis threshold' while diffing windows executable
I'm trying to diff two version of ntoskrnl.exe, using `rz-diff` 
So, when I run:
```
PS> rz-diff -AC -t functions .\ntoskrnl1.exe .\ntoskrnl2.exe
```
I get smth like this(and it goes on endlessly):
```
Exceeded analysis threshold while diffing fcn.1403d0fa0 and int.1403be8cc
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401c4d06
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403c4f77
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401bd75e
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403ce7cd
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403ce7e5
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403ce801
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403d443c
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401c7056
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1403d8d2e
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401d96be
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401cd9d6
Exceeded analysis threshold while diffing fcn.1403d0fa0 and fcn.1401cfbd2
Exceeded analysis threshold while diffing fcn.14048c540 and entry0
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_AlpcGetHeaderSize
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_AlpcGetMessageAttribute
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_AlpcInitializeMessageAttribute
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_CcCanIWrite
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_CcCoherencyFlushAndPurgeCache
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_CcCopyRead
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_CcCopyWrite
Exceeded analysis threshold while diffing fcn.14048c540 and sym.ntoskrnl.exe_CcCopyWriteWontFlush
```

I also tried to do this:
```
rz-diff -e analysis.depth=100 -AC -t functions .\ntoskrnl1.exe .\ntoskrnl2.exe
```
But it gave the same result.

/*My Versions*/
Rizin:
```
rizin 0.2.0-git @ windows-x86-64
commit: 6b3c58ebfaeb71a72737a2e60acd952f51ea96ed, build: Sun 03/14/2021__14:40:17.82
rz_diff 0.2.0-git @ windows-x86-64
commit: 6b3c58ebfaeb71a72737a2e60acd952f51ea96ed, build: Sun 03/14/2021__14:40:17.82
```

Windows:
`Version 1909 (OS Build 18363.418)`

Binaries:
I took them form this KB updates(x64): 
1) KB4471328 
2) KB4467107 

Attached them, for you to reproduce bug

[ntoskrnl.zip](https://github.com/rizinorg/rizin/files/6330533/ntoskrnl.zip)

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Resolves and speeds up the diffing of functions in 2 bins.
 
**Test plan**

already covered

**Closing issues**

https://github.com/rizinorg/rizin/issues/1044

### CODE REVIEW

File: test/db/tools/rz_diff [hunk 1]
```diff
@@ -200,38 +200,38 @@ sym.imp.__ctype_get_mb_cur_max    6 0x0000000000401150 | MATCH   (1.000000) | 0x
               sym.imp.iswprint    6 0x0000000000401370 | MATCH   (1.000000) | 0x0000000000401370     6 sym.imp.iswprint
          sym.imp.__ctype_b_loc    6 0x0000000000401380 | MATCH   (1.000000) | 0x0000000000401380     6 sym.imp.__ctype_b_loc
                   fcn.00401460   41 0x0000000000401460 | MATCH   (1.000000) | 0x0000000000401470    41 fcn.00401470
-                  fcn.00401530  745 0x0000000000401530 | UNMATCH (0.198093) | 0x0000000000401540   745 fcn.00401540
-                  fcn.004018f0  147 0x00000000004018f0 | UNMATCH (0.122449) | 0x0000000000401900   147 fcn.00401900
-                  fcn.00401990  137 0x0000000000401990 | UNMATCH (0.233577) | 0x00000000004019a0   137 fcn.004019a0
+                  fcn.00401530  745 0x0000000000401530 | SIMILAR (0.879195) | 0x0000000000401540   745 fcn.00401540
```
Comment (XVilka) [inline test/db/tools/rz_diff:203]: So essentially it as giving precisely the opposite of the valid result, right?

Comment (wargio) [reply test/db/tools/rz_diff:203]: yes

Comment (wargio) [reply test/db/tools/rz_diff:203]: before the double value was overwritten by the next compare. 
Specifically this line: https://github.com/rizinorg/rizin/pull/1346/files#diff-320081f996f71252b81a941f8e29742888f632f5da0ae5114a89cfd64d5a2b01L248

Comment (XVilka) [reply test/db/tools/rz_diff:203]: 🤦

**Review by XVilka — APPROVED** [review]

**Comment by wargio** [comment]
```
[XX] TIMEOUT db/asm/x86_64 <asm> mov rax, [rbx + 0xbeef]
[**]                                      db/asm/x86_64    11858 OK       801 BR        1 XX       15 FX
```

There seems to be some weird issue with rz-asm. this is not the first time i see something going in timeout

**Comment by XVilka** [comment]
I tested and confirm that it improves diffing significantly. Merging.

**Pull request merged**